src/server/helpers/utils.js

```diff
@@ -49,11 +49,119 @@ export function getEntityModels(orm: Object): Object {
 		Work
 	};
 }
+async function getParentAlias(orm, entityType, bbid) {
+	const rawSql = `
+		SELECT alias.name,
+			alias.sort_name,
+			alias.id,
+			alias.language_id,
+			alias.primary
+		FROM bookbrainz.${_.snakeCase(entityType)}
+		LEFT JOIN bookbrainz.alias ON alias.id = default_alias_id
+		WHERE bbid = '${bbid}' AND master = FALSE
+		ORDER BY revision_id DESC
+		LIMIT 1;
+	`;
+
+	// Query the database to get the parent revision default alias
+	const queryResult = await orm.bookshelf.knex.raw(rawSql);
+	if (!Array.isArray(queryResult.rows)) {
+		return null;
+	}
+	const rows = queryResult.rows.map((rawQueryResult) => {
+		// Convert query keys to camelCase
+		const queriedResult = _.mapKeys(
+			rawQueryResult,
+			(value, key) => _.camelCase(key)
+		);
+		return queriedResult;
+	});
+	return {
+		parentAlias: rows[0]
+	};
+}
+function getRevisionModels(orm) {
+	const {AuthorRevision, EditionRevision, EditionGroupRevision, PublisherRevision, WorkRevision} = orm;
+	return [
+		AuthorRevision,
+		EditionGroupRevision,
+		EditionRevision,
+		PublisherRevision,
+		WorkRevision
+	];
+}
+async function getCompleteRevision(revision, orm) {
+	const {Entity, AliasSet} = orm;
+	const RevisionModels = getRevisionModels(orm);
+	for (let i = 0; i < RevisionModels.length; i++) {
```

Comment:
OK, I tried a few things to end up with a pretty efficient way to fetch everything we need.

At the moment, you'll get a returned object that doesn't have the correct structure. I'll let you restructure that accordingly, I have to run away from my screen :)

```
for (let i = 0; i < RevisionModels.length; i++) {
	const Model = RevisionModels[i];
	const model = await new Model({id: revision.id})
		.fetch({
			require: false,
			withRelated: [
				'data.aliasSet.defaultAlias',
				'entity'
			]
		});

	if (model) {
		const modelJSON = model.toJSON();
		if (!modelJSON.dataId) {
			const bbid = modelJSON.bbid;
			// This feels a bit hacky, but shouldn't break: get e.g. 'edition_group' from 'bookbrainz.edition_group_revision' table name
			const modelType = model.tableName.match(/^bookbrainz.(\w+)_revision$/i)[1];
			const databaseModelName = _.upperFirst(_.camelCase(modelType));
			modelJSON.data.deleted = true;
			const parentAlias = await orm.func.entity.getEntityParentAlias(
				orm, databaseModelName, bbid
			);
			modelJSON.data.parentAlias = parentAlias;
		}

		const revisionId = revision.id;
		revision.editor = revision.author;
		delete revision.author;

		return {
			revisionId,
			...revision,
			...modelJSON
		};
	}
}
```

Note the withRelated: 'data.aliasSet.defaultAlias' ;  since we only need to load the alias, this nice shortcut using the ORM relationships saves us some hassle :)
See also the use of `orm.func.entity.getEntityParentAlias`

Reply:
I was try to use that getEntityParentAlias using import method but that didn't work. It worked but was showing some error, so I just copy pasted the function in the file for. I will change it

Reply:
orm.func.entity.getEntityParentAlias
I forgot I could use it this :P my bad

Reply:
Thanks sir. Yeah this is much better approach. I will restructure it accordingly.

Reply:
Oh, no need for honorifics !
I am no sir but a simple monkey :)